integration-app/src/main/java/uk/gov/crowncommercial/esourcing/integration/app/SalesforceConfiguration.java

```diff
@@ -1,24 +1,171 @@
+  private String clientSecret;
+
+  @Bean
+  ReactiveClientRegistrationRepository salesforceClientRegistrationRepository(
+      @Value("${spring.security.oauth2.client.provider.salesforce.token-uri}") String tokenUri,
+      @Value("${spring.security.oauth2.client.registration.salesforce.client-id}") String clientId,
+      @Value("${spring.security.oauth2.client.registration.salesforce.client-secret}")
+          String clientSecret) {
+    ClientRegistration salesforceClientRegistration =
+        ClientRegistration.withRegistrationId("salesforce")
+            .tokenUri(tokenUri)
+            .clientId(clientId)
+            .clientSecret(clientSecret)
+            .authorizationGrantType(AuthorizationGrantType.PASSWORD)
+            .build();
+    return new InMemoryReactiveClientRegistrationRepository(salesforceClientRegistration);
+  }
+
   @Bean
-  public ApiClient salesforceApiClient() {
+  public ApiClient salesforceApiClient(
+      ReactiveClientRegistrationRepository salesforceClientRegistrationRepository) {
+
+    ReactiveOAuth2AuthorizedClientService authorizedClientService =
+        new InMemoryReactiveOAuth2AuthorizedClientService(salesforceClientRegistrationRepository);
+
+    AuthorizedClientServiceReactiveOAuth2AuthorizedClientManager authorizedClientManager =
+        new AuthorizedClientServiceReactiveOAuth2AuthorizedClientManager(
+            salesforceClientRegistrationRepository, authorizedClientService);
+
+    WebClientReactivePasswordTokenResponseClient webClientReactivePasswordTokenResponseClient =
+        new WebClientReactivePasswordTokenResponseClient();
+
+    WebClient accessTokenWebClient =
+        WebClient.builder()
+            .filter(requestPasswordTokenRequestProcessor())
+            .filter(resposePasswordTokenRequestProcessor())
+            .build();
+
+    webClientReactivePasswordTokenResponseClient.setWebClient(accessTokenWebClient);
+
+    ReactiveOAuth2AuthorizedClientProvider authorizedClientProvider =
+        ReactiveOAuth2AuthorizedClientProviderBuilder.builder()
+            .password(
+                consumer ->
+                    consumer.accessTokenResponseClient(
+                        webClientReactivePasswordTokenResponseClient))
+            .build();
+
+    authorizedClientManager.setAuthorizedClientProvider(authorizedClientProvider);
 
-    ApiClient apiClient = new ApiClient();
+    authorizedClientManager.setContextAttributesMapper(
+        oAuth2AuthorizeRequest ->
+            Mono.just(
+                Map.of(
+                    OAuth2AuthorizationContext.USERNAME_ATTRIBUTE_NAME, username,
+                    OAuth2AuthorizationContext.PASSWORD_ATTRIBUTE_NAME, password)));
+
+    /*
+     * This code is largely duplicated from the autogenerated Salesforce API code
+     */
+    DateFormat dateFormat = new RFC3339DateFormat();
+    dateFormat.setTimeZone(TimeZone.getTimeZone("UTC"));
+    ObjectMapper mapper = new ObjectMapper();
```

Comment:
I think ObjectMapper creation is expensive - it does all sorts of classpath scanning. If this is configured in the same way as the other instance we have then best to pull out into the AppConfiguration (or even an ObjectMapperCOnfiguration) configuration class and then inject it into here. Will remove repeated code too.